src/authentication/authentication-hapi-plugin.ts

```diff
@@ -264,16 +296,16 @@ class AuthenticationHapiPlugin extends HapiPlugin {
   }
 
   private authorizeCustom(_userName: string, _request: Hapi.Request) {
-    return Promise.resolve(true);
+    return Promise.resolve(undefined);
   }
 
-  private validateFuncFactory(authorizer: (userName: string, request: Hapi.Request) => Promise<boolean>) {
+  private validateFuncFactory(authorizer: Authorizer) {
     return async (
       payload: AccessToken,
       request: Hapi.Request,
       callback: (err: any, valid: boolean, credentials?: any) => void,
     ) => {
-      let isAuthorized = false;
+      let isAuthorized: boolean | undefined = false;
```

Comment:
Since this can now have three values, I'd prefer e.g. an enum to make it explicit what each state means.